apps/api/src/tasks/bank-import/import-transactions.task.ts

```diff
@@ -0,0 +1,377 @@
+
+    const today = DateTime.now().toFormat('yyyy-MM-dd')
+
+    const response = (
+      await this.httpService.axiosRef.get<GetIrisTransactionInfoResponse>(
+        endpoint + `/${ibanAccount.id}` + `?dateFrom=${today}&dateTo=${today}`,
+        {
+          headers: {
+            'x-user-hash': this.userHash,
+            'x-agent-hash': this.agentHash,
+          },
+        },
+      )
+    ).data
+
+    return response.transactions
+  }
+
+  // Checks to see if all transactions have been processed already
+  private async hasNewOrNonImportedTransactions(transactions: IrisTransactionInfo[]) {
+    const today = new Date(DateTime.now().toFormat('yyyy-MM-dd'))
+
+    const count = await this.prisma.bankTransaction.count({
+      where: {
+        transactionDate: {
+          gte: today,
+          lte: today,
+        },
+      },
+    })
+
+    return transactions.length === count
+  }
+
+  // Only prepares the data, without inserting it in the DB
+  private prepareBankTransactionRecords(
+    transactions: IrisTransactionInfo[],
+    ibanAccount: IrisIbanAccountInfo,
+  ) {
+    const data: Prisma.BankTransactionCreateManyInput[] = transactions.map((trx) => ({
+      id: trx.transactionId?.trim() || ``,
+      ibanNumber: ibanAccount.iban,
+      bankName: ibanAccount.bankName,
+      bankIdCode: this.bankBIC,
+      transactionDate: new Date(trx.valueDate),
+      senderName: trx.debtorName,
+      recipientName: trx.creditorName,
+      senderIban: trx.debtorAccount?.iban,
+      recipientIban: trx.creditorAccount?.iban,
+      type: trx.creditDebitIndicator === 'CREDIT' ? 'credit' : 'debit',
+      amount: toMoney(trx.transactionAmount.amount),
+      currency: trx.transactionAmount.currency,
+      description: trx.remittanceInformationUnstructured.trim(),
+    }))
+
+    return data
+  }
+
+  private async processDonations(bankTransactions: Prisma.BankTransactionCreateManyInput[]) {
+    // Try to recognize campaign payment references
+    const matchedPaymentRef: string[] = []
+    bankTransactions.forEach((trx) => {
+      if (trx.type !== 'credit' || trx.description === 'STRIPE') return
+
+      const matchedRef = trx.description.replace(/[ _]+/g, '-').match(this.regexPaymentRef)
+      if (matchedRef) matchedPaymentRef.push(matchedRef[0])
+    })
+
+    /* 
+     Better get all campaigns in a single query
+     than execute a separate one for each transaction -
+     more performent and reliable approach
+    */
+    const campaigns = await this.prisma.campaign.findMany({
+      where: {
+        paymentReference: {
+          in: matchedPaymentRef,
+        },
+      },
+      include: {
+        vaults: true,
+      },
+    })
+
+    for (const trx of bankTransactions) {
+      // We're interested in parsing only incoming trx's
+      if (trx.type !== 'credit') {
+        continue
+      }
+
+      // Stripe payments should not be parsed
+      if (trx.description === 'STRIPE') {
+        continue
+      }
```

Comment:
this filter needs to be in the prepareBankTransactionRecords function, because even if filtering here, then on step 6 the credit transactions from STRIPE are still saved in the database, but they shouldn't  and must be skipped